USER_GUIDE.md

```diff
@@ -37,27 +40,81 @@ var client = new Client({
 });
 ```
 
-### To authenticate with the [Amazon OpenSearch Service](https://aws.amazon.com/opensearch-service/) use AwsSigv4Signer
+### To authenticate with [Amazon OpenSearch Service](https://aws.amazon.com/opensearch-service/) using AwsSigv4Signer
+
+#### AWS V3
 
 ```javascript
 const endpoint = ""; // OpenSearch domain URL e.g. https://search-xxx.region.es.amazonaws.com
 const { Client } = require('@opensearch-project/opensearch');
 const { AwsSigv4Signer } = require('@opensearch-project/opensearch/aws');
-const { defaultProvider } = require("@aws-sdk/credential-provider-node");
+const { STSClient, AssumeRoleCommand } = require("@aws-sdk/client-sts");
+
+async function assumeRole(roleArn, region) {
+  const client = new STSClient({ region });
+  const response = await client.send(
+    new AssumeRoleCommand({
+      RoleArn: roleArn,
+      RoleSessionName: "aws-es-connection",
+    })
+  );
+  return {
+    accessKeyId: response.Credentials.AccessKeyId,
+    secretAccessKey: response.Credentials.SecretAccessKey,
+    sessionToken: response.Credentials.SessionToken,
+  };
+}
 
 async function getClient() {
-  const credentials = await defaultProvider()();
-  var client = new Client({
-    ...AwsSigv4Signer({
-      credentials: credentials,
-      region: "us-west-2",
-    }),
+  const creds = await assumeRole(
+    "arn:aws:iam::0123456789012:role/Administrator",
+    "us-east-1"
+  );
+  const awsConfig = {
+    region: "us-east-1",
+    credentials: creds,
+  };
+  const client = new Client({
+    ...AwsSigv4Signer(awsConfig),
     node: endpoint,
   });
   return client;
 }
 ```
 
+#### AWS V2 global configuration
+
+```javascript
+const { Client } = require('@opensearch-project/opensearch');
+const { AwsSigv4Signer } = require('@opensearch-project/opensearch/aws');
+const AWS = require('aws-sdk');
+// (Optional) load profile credentials from file
+AWS.config.update({
+  profile: "my-profile",
+});
+const client = new Client({
+  ...AwsSigv4Signer(awsConfig),
```

Comment:
Can you update how a user can initialize the `awsConfig` object here? It isn't too clear from the example.

Reply:
Oops, good catch! this is a mistake it should be:
```suggestion
  ...AwsSigv4Signer(AWS.config),
```
There is two examples the first one (this highlighted) use the global AWS.config object (refer to the line just above the one you highlighted) and the second example shows how to instantiate an AWS.Config object specific to this service.
These examples are taken from the `aws-opensearch-connector` library.
I think we can remove the profile line as it is just confusing and this is not something AWS recommend using to obtain credentials in a live environment (even in a local this can be handled using aws-sdk ...PROFILE environment variable)
I will update that example to:
```javascript
const { Client } = require('@opensearch-project/opensearch');
const { AwsSigv4Signer } = require('@opensearch-project/opensearch/aws');
const AWS = require('aws-sdk');
const client = new Client({
  ...AwsSigv4Signer(AWS.config),
  node: endpoint,
});
```
which is less confusing.